plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java

```diff
@@ -4265,6 +4267,111 @@ else if (material.hasModernData() && material.getModernData() instanceof org.buk
+            // @returns LocationTag
+            // @warning See <@link language Structure lookups> for potential issues/edge cases in structure lookups.
+            // @group finding
+            // @description
+            // Finds the closest structure of the given type within the specified chunk radius (if any), optionally only searching for unexplored ones.
+            // For a list of default structures, see <@link url https://minecraft.wiki/w/Structure#ID>.
+            // Alternatively, you can specify a custom structure from a datapack, plugin, etc. as a namespaced key.
+            // See also <@link tag LocationTag.find_structure_type> to find structures by type,
+            // and <@link tag server.structures> for all structures currently available on the server.
+            // @example
+            // # Use to find the desert temple closest to the player, and tell them what direction it's in.
+            // - define found <player.location.find_structure[structure=desert_pyramid;radius=200].if_null[null]>
+            // - if <[found]> != null:
+            //   - narrate "The closet desert temple is <player.location.direction[<[found]>]> of you!"
+            // - else:
+            //   - narrate "No desert temple found."
+            // -->
+            tagProcessor.registerTag(LocationTag.class, MapTag.class, "find_structure", (attribute, object, input) -> {
+                ElementTag structureName = input.getRequiredObjectAs("structure", ElementTag.class, attribute);
+                ElementTag radius = input.getRequiredObjectAs("radius", ElementTag.class, attribute);
+                if (structureName == null || radius == null) {
+                    return null;
+                }
+                org.bukkit.generator.structure.Structure structure = Registry.STRUCTURE.get(Utilities.parseNamespacedKey(structureName.asString()));
+                if (structure == null) {
+                    attribute.echoError("Invalid structure specified: " + structureName + '.');
+                    return null;
+                }
+                if (!radius.isInt()) {
+                    attribute.echoError("Invalid radius '" + radius + " specified': must be a number.");
+                    return null;
+                }
+                ElementTag unexplored = input.getElement("unexplored", "false");
+                if (!unexplored.isBoolean()) {
+                    attribute.echoError("Invalid 'unexplored' value '" + unexplored + "' specified: must be a boolean.");
+                    return null;
+                }
+                StructureSearchResult searchResult = object.getWorld().locateNearestStructure(object, structure, radius.asInt(), unexplored.asBoolean());
+                return searchResult != null ? new LocationTag(searchResult.getLocation()) : null;
+            });
+
+            // <--[tag]
+            // @attribute <LocationTag.find_structure_type[type=<type>;radius=<#>(;unexplored=<true/{false}>)]>
```

Comment:
marking on github to avoid it going lost: per Discord discussion, probably better to exclude the `_type` stuff entirely for now, as it seems effectively just redundant/confusing

Reply:
Removed